tests/performance-tests/src/utils/publicApiService.ts

```diff
@@ -0,0 +1,290 @@
+import { RefinedResponse } from 'k6/http';
+import HttpClient from './httpClient';
+
+const applicationJsonHeaders = {
+  'Content-Type': 'application/json',
+};
+
+export const datSetQueryNodeTypes = ['criteria', 'and', 'or', 'not'] as const;
+
+export type DataSetQueryNodeType = typeof datSetQueryNodeTypes[number];
+
+export const dataSetQueryIdOperators = ['eq', 'notEq', 'in', 'notIn'] as const;
+
+export type DataSetQueryIdOperator = typeof dataSetQueryIdOperators[number];
+
+export const dataSetQueryComparableOperators = [
+  'eq',
+  'notEq',
+  'in',
+  'notIn',
+  'gt',
+  'gte',
+  'lt',
+  'lte',
+] as const;
+
+export type DataSetQueryComparableOperator = typeof dataSetQueryComparableOperators[number];
+
+export interface Publication {
+  id: string;
+  title: string;
+  slug: string;
+}
+
+interface DataSet {
+  id: string;
+  name: string;
+  content: string;
+  timePeriods: {
+    start: string;
+    end: string;
+  };
+  geographicLevels: string[];
+}
+
+export interface Paged<T> {
+  paging: {
+    page: number;
+    pageSize: number;
+    totalResults: number;
+    totalPages: number;
+  };
+  results: T[];
+}
+
+interface Location {
+  id: string;
+  level: string;
+  options: GeographicLevelLocations[];
+  label: string;
+  code: string;
+}
+
+interface GeographicLevelLocations {
+  [level: string]: Location[];
+}
+
+export interface DataSetMeta {
+  totalResults: number;
+  filters: [
+    {
+      id: string;
+      hint: string;
+      label: string;
+      options: [
+        {
+          id: string;
+          label: string;
+          isAggregate: boolean;
+        },
+      ];
+      name: string;
+    },
+  ];
+  indicators: [
+    {
+      id: string;
+      label: string;
+      unit: string;
+      name: string;
+      decimalPlaces: 0;
+    },
+  ];
+  locations: GeographicLevelLocations;
+  timePeriods: (TimePeriod & { label: string })[];
+}
+
+export interface DataSetQueryIdPredicate {
+  eq?: string;
+  notEq?: string;
+  in?: string[];
+  notIn?: string[];
+}
+
+export interface DataSetQueryComparablePredicate<T> {
+  eq?: T;
+  notEq?: T;
+  gte?: T;
+  gt?: T;
+  lte?: T;
+  lt?: T;
+  in?: T[];
+  notIn?: T[];
+}
+
+export interface DataSetQueryCriteria {
+  filters?: DataSetQueryIdPredicate;
+  locations?: DataSetQueryIdPredicate;
+  parentLocations?: DataSetQueryIdPredicate;
```

Comment:
Don't have this anymore. It's just `locations` now (which accepts codes or ids).

We also might want to run some tests using codes too to see how that performs?

Reply:
Oh cool, the codes are in now?  That'd be great to test.

Reply:
I'm compiling further things to test on https://dfedigital.atlassian.net/browse/EES-4190.  If you think of other things you'd like testing, feel free to pop them on this ticket.

Reply:
Removed now ta